packages/e2e/test/wallet/PersonalWallet/handle.test.ts

```diff
@@ -0,0 +1,254 @@
+  const coins = 10_000_000n; // number of coins to use in each transaction
+
+  const initPolicyId = async () => {
+    wallet = (await getWallet({ env, idx: 0, logger, name: 'Handle Init Wallet', polling: { interval: 50 } })).wallet;
+    await walletReady(wallet, coins);
+    const derivationPath = {
+      index: 2,
+      role: KeyRole.External
+    };
+    const keyAgent = await createStandaloneKeyAgent(
+      env.KEY_MANAGEMENT_PARAMS.mnemonic.split(' '),
+      await firstValueFrom(wallet.genesisParameters$),
+      await wallet.keyAgent.getBip32Ed25519()
+    );
+    const pubKey = await keyAgent.derivePublicKey(derivationPath);
+    const keyHash = await keyAgent.bip32Ed25519.getPubKeyHash(pubKey);
+    policySigner = new util.KeyAgentTransactionSigner(keyAgent, derivationPath);
+    policyScript = {
+      __type: Cardano.ScriptType.Native,
+      kind: Cardano.NativeScriptKind.RequireAllOf,
+      scripts: [
+        {
+          __type: Cardano.ScriptType.Native,
+          keyHash,
+          kind: Cardano.NativeScriptKind.RequireSignature
+        }
+      ]
+    };
+    policyId = nativeScriptPolicyId(policyScript);
+    wallet.shutdown();
+  };
+
+  const mint = async (tokens: Cardano.TokenMap, txMetadatum: Cardano.Metadatum) => {
+    const auxiliaryData = {
+      blob: new Map([[721n, txMetadatum]])
+    };
+
+    const txProps: InitializeTxProps = {
+      auxiliaryData,
+      mint: tokens,
+      outputs: new Set([
+        {
+          address: walletAddress,
+          value: {
+            assets: tokens,
+            coins
+          }
+        }
+      ]),
+      witness: { extraSigners: [policySigner], scripts: [policyScript] }
+    };
+
+    const unsignedTx = await wallet.initializeTx(txProps);
+
+    const finalizeProps: FinalizeTxProps = {
+      auxiliaryData,
+      tx: unsignedTx,
+      witness: { extraSigners: [policySigner], scripts: [policyScript] }
+    };
+
+    const signedTx = await wallet.finalizeTx(finalizeProps);
+    await submitAndConfirm(wallet, signedTx);
+  };
+
+  const restartWallet = async () => {
+    wallet.shutdown();
+    wallet = (
+      await getWallet({
+        env,
+        handlePolicyIds: [policyId],
+        idx: 0,
+        logger,
+        name: 'Minting Wallet',
+        polling: { interval: 50 }
+      })
+    ).wallet;
+    await walletReady(wallet, coins);
+  };
+
+  beforeAll(async () => {
+    await initPolicyId();
+    wallet = (
+      await getWallet({
+        env,
+        handlePolicyIds: [policyId],
+        idx: 0,
+        logger,
+        name: 'Minting Wallet',
+        polling: { interval: 50 }
+      })
+    ).wallet;
+    await walletReady(wallet, coins);
+    receivingWallet = (
+      await getWallet({
+        env,
+        handlePolicyIds: [policyId],
+        idx: 1,
+        logger,
+        name: 'Receiving Wallet',
+        polling: { interval: 50 }
+      })
+    ).wallet;
+    await walletReady(wallet, coins);
```

Comment:
Here probably we want to wait for `receivingWallet` to be ready rather than for `wallet`.

One more suggestions: grouping the two calls as follows should speed up a bit the test
```
await Promise.all([walletReady(wallet, coins), walletReady(receivingWallet, coins)]);
```